File: workout-logger-backend/models/WorkoutLog.js
const mongoose = require('mongoose');

const WorkoutLogSchema = new mongoose.Schema({
  userId: { type: String, required: true },
  exerciseId: { type: mongoose.Schema.Types.ObjectId, ref: 'Exercise', required: true },
  weight: { type: Number, required: true },
  reps: { type: Number, required: true },
  totalSets: { type: Number }, // ✅ New field
  date: { type: Date, default: Date.now },
  notes: { type: String },
});

WorkoutLogSchema.index({ exerciseId: 1, date: -1 });


module.exports = mongoose.model('WorkoutLog', WorkoutLogSchema);

File: workout_logger_app/lib/models/workout_log.dart
class WorkoutLog {
  final String id;
  final String userId;
  final String exerciseId;
  final double weight;
  final int reps;
  final String? notes;
  final DateTime date;
  final int totalSets;

  WorkoutLog({
    required this.id,
    required this.userId,
    required this.exerciseId,
    required this.weight,
    required this.reps,
    this.notes,
    required this.date,
    required this.totalSets,
  });

  factory WorkoutLog.fromJson(Map<String, dynamic> json) {
    return WorkoutLog(
      id: json['_id'],
      userId: json['userId'],
      exerciseId: json['exerciseId'],
      weight: (json['weight'] as num).toDouble(),
      reps: json['reps'],
      notes: json['notes'],
      date: DateTime.parse(json['date']),
      totalSets: json['totalSets'],
    );
  }
}
